openhabian-setup.sh

```diff
@@ -327,42 +359,39 @@ misc_system_settings() {
   echo -n "[openHABian] Applying multiple useful system settings (permissions, ...)... "
   cond_redirect adduser openhab dialout
   cond_redirect adduser openhab tty
-  cond_redirect adduser pi openhab
-  cond_redirect adduser pi dialout
-  cond_redirect adduser pi tty
+  cond_redirect adduser $username openhab
+  cond_redirect adduser $username dialout
+  cond_redirect adduser $username tty
   cond_redirect setcap 'cap_net_raw,cap_net_admin=+eip cap_net_bind_service=+ep' `realpath /usr/bin/java`
-  cond_redirect chown -R openhab:pi /opt /etc/openhab2
+  cond_redirect chown -R openhab:$username /opt /etc/openhab2
   cond_redirect chmod -R g+w /opt /etc/openhab2
   echo "OK"
 }
 
-openhab_shell_interfaces() {
+change_karaf_default_username(){
+  echo -n "[openHABian] Changing karaf user from \"openhab\" to \"$username\"... "
+  # Enable encryption of password within openhab2 karaf console
+  cond_redirect sed -i "s/encryption.enabled = false/encryption.enabled = true/g" /var/lib/openhab2/etc/org.apache.karaf.jaas.cfg
+  cond_redirect sed -i "s/encryption.name =/encryption.name = basic/g" /var/lib/openhab2/etc/org.apache.karaf.jaas.cfg
+  cond_redirect sed -i "s/encryption.algorithm = MD5/encryption.algorithm = SHA-256/g" /var/lib/openhab2/etc/org.apache.karaf.jaas.cfg
```

Comment:
Is there any drawback of these settings? If not, why not harden the setup for everybody (instead of treating it as a post install issue).
The Karaf files of the distro are maintained here: https://github.com/openhab/openhab-distro/tree/master/distributions/openhab/src/main/resources/userdata/etc